lib/audited/auditor.rb

```diff
@@ -250,18 +250,18 @@ def write_audit(attrs)
         end
       end
 
-      def presence_of_audit_comment
-        if comment_required_state?
-          errors.add(:audit_comment, "Comment can't be blank!") unless audit_comment.present?
+      def eligible_for_comment_validation?
+        if !auditing_enabled ||
+          (audited_options[:on].exclude?(:create) && self.new_record?) ||
+          (audited_options[:on].exclude?(:update) && self.persisted?) ||
+          (audited_changes.empty? && self.persisted?)
+        then
+          false
+        else
+          true
```

Comment:
This is quite a confusing check, returning the opposite result to the condition - perhaps you can just reuse the `comment_required_state?` function and add the additional condition instead of flipping the entire logic around?

Reply:
I think it was mainly just the name `comment_required_state?` that I wasn't a fan of. To me it seems to indicate whether or not `comment_required` is enabled, but what it's really doing is adding additional checks for whether or not `audit_comment` should be validated for presence when `comment_required` is enabled. I felt that `eligible_for_comment_validation?` was clearer about what it's actually checking.

And my though process for flipping the logic around a bit was just so that the conditions would fail earlier – meaning that if auditing isn't enabled, we can immediately return `false` instead of having to test any more conditions. But I can definitely understand how it's a bit confusing. And now that I think about it, my logic was complete non-sense anyways – `if !auditing_enabled ||` and `auditing_enabled &&` will both return false immediately if auditing isn't enabled. So that was just a big derp on my part. 😬 

I'll get this fixed today.